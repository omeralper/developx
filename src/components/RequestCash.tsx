import React from "react";
import { requestCashAction, requestCashActionFunction } from "../actions/CashActions";
import { ICashResponse } from "../api/CashApi";
import { connect } from "react-redux";

export interface IRequestCashProps {
    requestCashAction: requestCashActionFunction;
}

interface IRequestCashState {
    cashAmount: number;
}

export class RequestCash extends React.PureComponent<IRequestCashProps,IRequestCashState> {
    constructor(props: IRequestCashProps) {
        super(props);
        this.state = {
            cashAmount: 0
        };
    }

    onCashSubmit = () => { 
        this.props.requestCashAction(this.state.cashAmount);
    }

    setCashAmount = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({
            cashAmount: +event.target.value
         })  
    }

    render() {
        return (
            <div className="RequestCash">
               <form className="text-center border border-light p-5">
                    <p className="h4 mb-4">Withdraw Cash</p>
                    <input type="number" className="form-control mb-4" placeholder="Amount" onChange={this.setCashAmount} />
                    <input type="buton" className="btn btn-info my-4" onClick={this.onCashSubmit} value="Submit"/>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = {
    requestCashAction
}

export default connect(null,mapDispatchToProps)(RequestCash);