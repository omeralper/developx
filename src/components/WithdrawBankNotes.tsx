import React from "react";
import { ICashResponseEntity } from "../api/CashApi";
import { connect } from "react-redux";
import { ICashReducerState } from "../reducers/CashReducer";

export interface IWithdrawBankNotesProps {
    withdrawMoneySet: ICashResponseEntity[],
    notAvailable: boolean
}

export const WithdrawBankNotes: React.StatelessComponent<IWithdrawBankNotesProps> = ({
    withdrawMoneySet,
    notAvailable
}) => {
    return (
        <div className="WithdrawBankNotes">
         {notAvailable ? "Not Available" : 
            <ul>
                {withdrawMoneySet.map((cashEntity, i) => (
                    <li key={i}>Bank note {cashEntity.bankNote} - Amount  {cashEntity.amount}</li>
                ))}
            </ul>
         }
        </div>
    );
}

const mapStateToProps = ({withdrawMoneySet,notAvailable}: ICashReducerState) => ({
    withdrawMoneySet,
    notAvailable
})

export default connect(mapStateToProps)(WithdrawBankNotes);