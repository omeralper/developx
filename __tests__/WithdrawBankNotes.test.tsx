import * as React from "react";
import * as renderer from "react-test-renderer";
import {WithdrawBankNotes, IWithdrawBankNotesProps } from "../src/components/WithdrawBankNotes";
import { shallow } from "enzyme";

let props = {} as IWithdrawBankNotesProps;

describe("Withdraw Bank Notes Component", () => {
    beforeEach(() =>{
        props = {} as IWithdrawBankNotesProps;
    });

    it("shoud render withdrawbanknotes component properly", () => {
        props.withdrawMoneySet = [];
        const component = renderer.create(<WithdrawBankNotes {...props} />);
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("shoud render 2 list item for 250 ", () => {
        props.withdrawMoneySet =  [
                {
                    bankNote : 100,
                    amount: 2
                },
                {
                    bankNote : 50,
                    amount: 1
                }
            ];
        const wrapper = shallow(<WithdrawBankNotes {...props}/>);
        const listItems = wrapper.find('li');
        expect(listItems.length).toBe(2);
    });
})