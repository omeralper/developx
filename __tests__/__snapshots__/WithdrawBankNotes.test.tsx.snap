// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Withdraw Bank Notes Component shoud render withdrawbanknotes component properly 1`] = `
<div
  className="WithdrawBankNotes"
>
  <ul />
</div>
`;
